frontend/mainwindow.cpp

```diff
@@ -318,4 +337,10 @@ void MainWindow::setLoadScreen(bool state)
         QWidget::setEnabled(true);
         QGuiApplication::restoreOverrideCursor();
     }
-}
\ No newline at end of file
+}
+
+void MainWindow::on_actionRB_No_triggered() { massSearch("RB No.", "run_number"); }
+
+void MainWindow::on_actionTitle_triggered() { massSearch("Title", "title"); }
+
+void MainWindow::on_actionUser_triggered() { massSearch("User name", "user_name"); }
```

Comment:
Giving these actions more explicit names (e.g. `actionMassSearchTitle`) would aid code readability and understanding.